src/labkey/Storage.ts

```diff
@@ -0,0 +1,224 @@
+ *                  console.log(response);
+ *              }
+ *          });
+ *      }
+ * });
+ * ```
+ *
+ * ```js
+ * // Create a terminal storage location in the freezer
+ * LABKEY.Storage.createStorageItem({
+ *      type: 'Storage Unit Type',
+ *      props: {
+ *          name: '10 X 10 Box',
+ *          unitType: 'Box',
+ *          rows: 10,
+ *          cols: 10
+ *      },
+ *      success: function(response) {
+ *          console.log(response);
+ *          var boxTypeId = response.data.rowId;
+ *
+ *          LABKEY.Storage.createStorageItem({
+ *              type: 'Terminal Storage Location',
+ *              props: {
+ *                  name: 'Box #1',
+ *                  typeId: boxTypeId,
+ *                  locationId: 8088, // rowId of Shelf #2
+ *              },
+ *              success: function(response) {
+ *                  console.log(response);
+ *              }
+ *          });
+ *      }
+ * });
+ * ```
+ */
+export function createStorageItem(config: IStorageCommandOptions): XMLHttpRequest {
+    return request({
+        url: buildURL('storage', 'create.api', config.containerPath),
+        method: 'POST',
+        jsonData: {
+            type: config.type,
+            props: config.props,
+        },
+        success: getCallbackWrapper(getOnSuccess(config), config.scope),
+        failure: getCallbackWrapper(getOnFailure(config), config.scope, true),
+    });
+}
+
+/**
+ * Update an existing LabKey Freezer Manager storage item to change its properties or location within the freezer hierarchy.
+ * For the UpdateCommand, the "rowId" primary key value is required to be set within the props.
+ *
+ * ```js
+ * // Update the properties of a freezer
+ * LABKEY.Storage.updateStorageItem({
+ *      type: 'Freezer',
+ *      props: {
+ *          rowId: 8087,
+ *          description: 'Updated freezer from API',
+ *          status: 'Defrosting'
+ *      },
+ *      success: function(response) {
+ *          console.log(response);
+ *      }
+ * });
+ * ```
+ *
+ * ```js
+ * // Update the location of a box in the freezer
+ * LABKEY.Storage.updateStorageItem({
+ *      type: 'Terminal Storage Location',
+ *      props: {
+ *          rowId: 19382,
+ *          locationId: 8089 // move Box #1 from Shelf #1 to Shelf #2
+ *      },
+ *      success: function(response) {
+ *          console.log(response);
+ *      }
+ * });
+ * ```
+ */
+export function updateStorageItem(config: IStorageCommandOptions): XMLHttpRequest {
+    return request({
+        url: buildURL('storage', 'update.api', config.containerPath),
+        method: 'POST',
+        jsonData: {
+            type: config.type,
+            props: config.props,
+        },
+        success: getCallbackWrapper(getOnSuccess(config), config.scope),
+        failure: getCallbackWrapper(getOnFailure(config), config.scope, true),
+    });
+}
+
+export interface IDeleteStorageCommandOptions extends IStorageCommandOptions {
```

Comment:
We don't use the `I` prefix on interfaces any longer.

Reply:
done